pkg/apiserver/user/sqlauth/sqlauth.go

```diff
@@ -60,3 +147,22 @@ func (a *Authenticator) Authenticate(f user.AuthenticateForm) (*utils.SessionUse
 		IsWriteable:  true,
 	}, nil
 }
+
+// grantRows examples:
+// - GRANT PROCESS,SHOW DATABASES,CONFIG ON *.* TO 'dashboardAdmin'@'%'
+// - GRANT SYSTEM_VARIABLES_ADMIN,RESTRICTED_VARIABLES_ADMIN,RESTRICTED_STATUS_ADMIN,RESTRICTED_TABLES_ADMIN ON *.* TO 'dashboardAdmin'@'%'
+// - GRANT ALL PRIVILEGES ON *.* TO 'dashboardAdmin'@'%'
+func parseGrants(grantRows []string) []string {
+	grants := make([]string, 0)
+	grantRegex := regexp.MustCompile(`GRANT (.+) ON`)
+
+	for _, row := range grantRows {
+		m := grantRegex.FindStringSubmatch(row)
+		if len(m) == 2 {
+			curRowGrants := strings.Split(m[1], ",")
```

Comment:
I feel highly unsafe about this process logic. We need to confirm how TiDB may possibly outputs these grants.

Reply:
It's ok here. TiDB is using `,` to separate the privilege.

https://github.com/pingcap/tidb/blob/master/privilege/privileges/cache.go#L1322